common/djangoapps/third_party_auth/provider.py

```diff
@@ -103,9 +104,11 @@ def get_enabled_by_backend_name(cls, backend_name):
             Instances of ProviderConfig.
         """
         if backend_name in _PSA_OAUTH2_BACKENDS:
-            provider = OAuth2ProviderConfig.current(backend_name)
-            if provider.enabled_for_current_site:
-                yield provider
+            oauth2_slugs = OAuth2ProviderConfig.key_values('provider_slug', flat=True)
+            for oauth2_slug in oauth2_slugs:
+                provider = OAuth2ProviderConfig.current(oauth2_slug)
+                if provider.enabled_for_current_site and provider.backend_name in _PSA_OAUTH2_BACKENDS:
```

Comment:
Yeah you are right, this would have left it returning too easily. I fixed this and added a test that checks against the bug this would have created, thanks!